hs_file_types/models/base.py

```diff
@@ -852,22 +906,34 @@ def _validate_set_file_type_inputs(cls, resource, file_id=None, folder_path=''):
                 raise ValidationError(msg)
 
             # check if a FileSet aggregation can be created from the specified folder
-            if not resource.can_set_folder_to_fileset(path_to_check):
-                msg = "FileSet aggregation can't be created from the specified folder:{}"
-                msg = msg.format(path_to_check)
-                raise ValidationError(msg)
-
-        if cls.__name__ != 'FileSetLogicalFile':
+            if cls.__name__ == "FileSetLogicalFile":
+                if not resource.can_set_folder_to_fileset(path_to_check):
```

Comment:
If we move the logical file type validation to the logical files and away from the resource, we can just run a single method on the logical file class.